app/services/metrc_service/plant/harvest.rb

```diff
@@ -74,10 +74,21 @@ def build_remove_waste_payload
         end
       end
 
-      def waste_type(_completion)
-        # TODO: determine waste type from 'process' completion
-
-        'Plant Material'
+      # If the waste type (the resource unit's +product_modifier+) is found in the list of available waste types
+      # from Metrc (GET [metrc]/harvests/v1/waste/types), # then we will use that.
+      #
+      # If not, then we use the first available valid waste type from Metrc
+      def waste_type(completion)
```

Comment:
@sd @betacar could you please check this method and let me know if this aligns with your thinking